plugin/src/main/java/org/opensearch/ml/cluster/MLCommonsClusterManagerEventListener.java

```diff
@@ -62,13 +70,28 @@ public MLCommonsClusterManagerEventListener(
 
     @Override
     public void onClusterManager() {
-        if (syncModelRoutingCron == null) {
-            startSyncModelRoutingCron();
-        }
+        ActionListener<Boolean> listener = ActionListener.wrap(r -> {
+            if (syncModelRoutingCron == null) {
+                startSyncModelRoutingCron();
+            }
+        }, e -> {
+            if (syncModelRoutingCron == null) {
+                startSyncModelRoutingCron();
+            }
+        });
+        mlModelAutoReDeployer.setStartCronJobListener(listener);
+        String localNodeId = clusterService.localNode().getId();
+        threadPool
+            .schedule(
+                () -> mlModelAutoReDeployer.buildAutoReloadArrangement(List.of(localNodeId), localNodeId),
+                TimeValue.timeValueSeconds(jobInterval),
```

Comment:
Can we start auto reload immediately ?

Reply:
No, cluster manager ready doesn't mean cluster is ready, if we start auto reload immediately we can get a `cluster not initialized` etc exception when querying needs reload models.